sky/shell/platform/android/org/domokit/sky/shell/FlutterSemanticsToAndroidAccessibilityBridge.java

```diff
@@ -0,0 +1,279 @@
+            persistentNode.update(node);
+        } else {
+            persistentNode = new PersistentAccessibilityNode(node);
+            mTreeNodes.put(node.id, persistentNode);
+        }
+        assert persistentNode != null;
+        return persistentNode;
+    }
+
+    public void removePersistentNode(PersistentAccessibilityNode node) {
+        assert mTreeNodes.containsKey(node.id);
+        assert mTreeNodes.get(node.id).parent == null;
+        mTreeNodes.remove(node.id);
+    }
+
+    public void reset() {
+        mTreeNodes.clear();
+    }
+
+    private class PersistentAccessibilityNode {
+        PersistentAccessibilityNode(SemanticsNode node) {
+            update(node);
+        }
+        void update(SemanticsNode node) {
+            if (id == -1) {
+                id = node.id;
+                assert node.flags != null;
+                assert node.strings != null;
+                assert node.geometry != null;
+                assert node.children != null;
+            }
+            assert id == node.id;
+            if (node.flags != null) {
+                canBeTapped = node.flags.canBeTapped;
+                canBeLongPressed = node.flags.canBeLongPressed;
+                canBeScrolledHorizontally = node.flags.canBeScrolledHorizontally;
+                canBeScrolledVertically = node.flags.canBeScrolledVertically;
+                hasCheckedState = node.flags.hasCheckedState;
+                isChecked = node.flags.isChecked;
+            }
+            if (node.strings != null) {
+                label = node.strings.label;
+            }
+            if (node.geometry != null) {
+                transform = node.geometry.transform;
+                left = node.geometry.left;
+                top = node.geometry.top;
+                width = node.geometry.width;
+                height = node.geometry.height;
+            }
+            if (node.children != null) {
+                List<PersistentAccessibilityNode> oldChildren = children;
+                children = new ArrayList<PersistentAccessibilityNode>(node.children.length);
+                if (oldChildren != null) {
+                    for (PersistentAccessibilityNode child : oldChildren) {
+                        assert child.parent != null;
+                        child.parent = null;
+                    }
+                }
+                for (SemanticsNode childNode : node.children) {
+                    PersistentAccessibilityNode child = FlutterSemanticsToAndroidAccessibilityBridge.this.updateSemanticsNode(childNode);
+                    assert child != null;
+                    child.parent = this;
+                    children.add(child);
+                }
+                if (oldChildren != null) {
+                    for (PersistentAccessibilityNode child : oldChildren) {
+                        if (child.parent == null) {
+                            FlutterSemanticsToAndroidAccessibilityBridge.this.removePersistentNode(child);
```

Comment:
Not if the children of the orphaned parent are still referenced from mTreeNodes, right?